tools/maven-bundle-plugin/src/main/java/org/apache/felix/bundleplugin/ManifestPlugin.java

```diff
@@ -428,15 +428,27 @@ private boolean isMetadataUpToDate(File outputFile, MavenProject project)
             // generated last?
             Path cacheData = getIncrementalDataPath(project);
             if(!Files.isRegularFile(cacheData)) {
+                getLog().debug("No cache data file found at '" + cacheData + "', generating manifest.");
                 return false;
             }
             long manifestLastModified = lastModified(cacheData);
             while (project != null)
             {
-                Path pom = project.getFile().toPath();
-                if (manifestLastModified < lastModified(pom))
-                {
-                    return false;
+                if (project.getFile() != null) {
+                    Path pom = project.getFile().toPath();
+                    if (manifestLastModified < lastModified(pom))
+                    {
+                        getLog().debug("File at  '" + pom + "' newer than cache data file, generating manifest.");
+                        return false;
+                    }
+                } else {
+                    if (project.getVersion().endsWith("-SNAPSHOT")) { // is it mutable?
+                        getLog().debug("pom.xml file not found for SNAPSHOT project'" + project + "', assume modification.");
+                        return false;
+                    } else {
+                        getLog().debug("pom.xml file not found for non-SNAPSHOT project'" + project + "', assume no modification.");
+                        break;
```

Comment:
If it is a release version all its parents must be non-SNAPSHOTS as well. A change would require a version change which would propagate down to all ancestor poms

Reply:
That is true for jars one fetches from Maven-Central, but if you are in the IDE and are going back and forth in the git-history weird scenarios can happen. Therefore I think it would be better to just continue.

Reply:
Ok, done in https://github.com/apache/felix-dev/pull/169/commits/430c03299f3a747dbcc629e32f9d1f7addf7a32f.

Reply:
Thank you.